src/uta/parsers/seqrecord.py

```diff
@@ -63,25 +118,62 @@ def gene_id(self):
         assert len(gene_ids) == 1
         return gene_ids[0]
 
+    @property
+    def db_xrefs(self):
+        """
+         gene            1..4577
+                 /gene="A2M"
+                 /gene_synonym="DKFZp779B086; FWP007; S863-7"
+                 /db_xref="GeneID:2"
+                 /db_xref="HPRD:00072"
+                 /db_xref="MIM:103950"
+        """
+        db_xrefs = self.gene_feature.qualifiers["db_xref"]
+        return [xref for xref in db_xrefs]
+
     @property
     def cds_se_i(self):
-        if self.features_by_type.get("CDS"):
-            cds_feature = self.features_by_type.get("CDS")[0]
-            return (cds_feature.location.start.real, cds_feature.location.end.real)
+        if self.cds_feature is not None:
+            return self.cds_feature.location.start.real, self.cds_feature.location.end.real
+        else:
+            return None
+
+    @property
+    def cds_product(self):
+        if self.cds_feature is not None:
+            return self.cds_feature.qualifiers["product"][0]
+        else:
+            return None
+
+    @property
+    def cds_protein_id(self):
+        if self.cds_feature is not None:
+            return self.cds_feature.qualifiers["protein_id"][0]
+        else:
+            return None
+
+    @property
+    def cds_translation(self):
+        if self.cds_feature is not None:
+            return str(self.cds_feature.qualifiers["translation"][0])
         else:
             return None
 
     @property
     def exons_se_i(self):
+        se_i = []
         if "exon" in self.features_by_type:
             exons = self.features_by_type["exon"]
-        else:
-            exons = [self.gene_feature]
```

Comment:
It turns out that if "exon" is not defined as a feature in the genbank record, we cannot assume it is the same as "gene" or any other feature that I can find.